src/RateTable1.java

```diff
@@ -0,0 +1,127 @@
+import java.math.BigDecimal;
+
+import components.map.Map;
+import components.map.Map.Pair;
+import components.map.Map1L;
+
+/**
+ * {@code RateTable} represented as a {@code Map} with implementations of
+ * primary methods.
+ *
+ * @convention <pre>
+ * [the RateTable map is non-null, contains unique keys (currency names),
+ *             and each associated BigDecimal value represents a valid exchange rate]
+ * </pre>
+ * @correspondence <pre>
+ * this = [map of currency names (String) to exchange rate values (BigDecimal)]
+ * </pre>
+ *
+ * @author Ryan Ouimet
+ *
+ */
+public class RateTable1 extends RateTableSecondary {
+
+    /*
+     * Private members --------------------------------------------------------
+     */
+
+    /**
+     * Representation of {@code this}.
+     */
+    private Map<String, BigDecimal> rep;
```

Comment:
Do you not plan to store the exchange rates at all here? I actually think this is fine. Then, you return Exchange rates and not the mutable Pair. Really good stuff.